Userland/Libraries/LibDebug/Dwarf/DwarfInfo.h

```diff
@@ -7,14 +7,15 @@
 #pragma once
 
 #include "AttributeValue.h"
-#include "CompilationUnit.h"
 #include "DwarfTypes.h"
 #include <AK/ByteBuffer.h>
 #include <AK/NonnullOwnPtrVector.h>
 #include <AK/NonnullRefPtr.h>
 #include <AK/RedBlackTree.h>
 #include <AK/RefCounted.h>
 #include <AK/String.h>
+#include <LibDebug/Dwarf/CompilationUnit.h>
+#include <LibDebug/Dwarf/DIE.h>
```

Comment:
These don't seem needed to be included in the header, as their types are used only as pointers & template params?

Reply:
They are needed. The compiler says so, and for a good reason. DwarfInfo contains
```
    NonnullOwnPtrVector<Dwarf::CompilationUnit> m_compilation_units;
```
and an implicit destructor, which needs to be able to call the destructor of CompilationUnit. The latter is impossible with an incomplete definition of CompilationUnit.

You're right about DIE.h being unnecessary; it is already included by CompilationUnit.

Reply:
> You're right about DIE.h being unnecessary; it is already included by CompilationUnit.

I wouldn't remove it for that reason, implicit includes are not good.

IMO a out-of-line destructor is the way to go here as well.

Reply:
Okay, I'll keep the include.

Why put the destructor out of line? The cyclic includes are already fixed by LineProgram no longer including DwarfInfo.

If you want to remove the CompilationUnit include, then that will probably break `DwarfInfo::for_each_compilation_unit` in weird ways.